presto-tests/src/main/java/com/facebook/presto/tests/H2QueryRunner.java

```diff
@@ -374,6 +375,10 @@ private static Object[] mapArrayValues(ArrayType arrayType, Object[] values)
                     .toArray();
         }
 
+        if (elementType instanceof DateType) {
```

Comment:
Yes. This is not ported, it's added for this PR.
This is because the tests added include Date type. In our test code, we convert the returned result for both Presto and H2 query runner to the same type for comparison. This code is to handle the case for DateType.